tests/rptest/tests/random_node_operations_test.py

```diff
@@ -84,11 +84,6 @@ def _setup_test_scale(self, num_to_upgrade):
         self.producer_timeout = 180
         self.consumer_timeout = 180
 
-        if num_to_upgrade > 0:
-            # we can only enable controller snapshot if all nodes are the newest version
```

Comment:
Hmm, why is this check deleted? Looks like it got a bit mangled in the previous PRs but it should still be somewhere - a test with both upgrade and controller snapshots won't work

Reply:
this check was moved to different place, we simply return a flag that indicates if a test should be skipped.

Reply:
AFAICS we will still run the test with the following combination of parameters: dedicated nodes + controller snapshots + upgrade, and this is going to fail.

Reply:
you are right, fixed